# Polycystic Ovary Syndrome | Dataset

Polycystic Ovary Syndrome (PCOS) is a hormonal imbalance disorder affecting women of reproductive age. 
It is characterized by irregular menstrual cycles, excess androgen (male hormone) levels, and the presence 
of multiple small cysts in the ovaries.
Women with PCOS may experience various symptoms, including weight gain, acne, and difficulty getting pregnant.

In [2]:
#importing libraries

import numpy as np
import pandas as pd

# Load the dataset into Jupyter Notebook

There are various formats for a dataset, .csv, .json, .xlsx  etc. The dataset can be stored in different places, on your local machine or sometimes online. In our case, the PCOS Dataset is an online source, and it is in CSV (comma separated value) format.

dataset name : PCOS_data.csv

The Pandas Library is a useful tool that enables us to read various datasets into a data frame; our Jupyter notebook platforms have a built-in Pandas Library so that all we need to do is import Pandas without installing.

In [3]:
#load the dataset
#Dataset source - https://www.kaggle.com/datasets/shreyasvedpathak/pcos-dataset
#Read the dataset in pcos varaible

pcos = pd.read_csv('PCOS_data.csv')
pcos

Sl. No  Patient File No.  PCOS (Y/N)   Age (yrs)  Weight (Kg)  \
0         1                 1           0          28         44.6   
1         2                 2           0          36         65.0   
2         3                 3           1          33         68.8   
3         4                 4           0          37         65.0   
4         5                 5           0          25         52.0   
..      ...               ...         ...         ...          ...   
536     537               537           0          35         50.0   
537     538               538           0          30         63.2   
538     539               539           0          36         54.0   
539     540               540           0          27         50.0   
540     541               541           1          23         82.0   

     Height(Cm)    BMI  Blood Group  Pulse rate(bpm)   RR (breaths/min)  ...  \
0        152.000  19.3           15                78                22  ...   
1        161.500  24.9           15                74                20  ...   
2        165.000  25.3           11                72                18  ...   
3        148.000  29.7           13                72                20  ...   
4        161.000  20.1           11                72                18  ...   
..           ...   ...          ...               ...               ...  ...   
536      164.592  18.5           17                72                16  ...   
537      158.000  25.3           15                72                18  ...   
538      152.000  23.4           13                74                20  ...   
539      150.000  22.2           15                74                20  ...   
540      165.000  30.1           13                80                20  ...   

     Fast food (Y/N)  Reg.Exercise(Y/N)  BP _Systolic (mmHg)  \
0                1.0                  0                  110   
1                0.0                  0                  120   
2                1.0                  0                  120   
3                0.0                  0                  120   
4                0.0                  0                  120   
..               ...                ...                  ...   
536              0.0                  0                  110   
537              0.0                  0                  110   
538              0.0                  0                  110   
539              0.0                  0                  110   
540              1.0                  0                  120   

     BP _Diastolic (mmHg)  Follicle No. (L)  Follicle No. (R)  \
0                      80                 3                 3   
1                      70                 3                 5   
2                      80                13                15   
3                      70                 2                 2   
4                      80                 3                 4   
..                    ...               ...               ...   
536                    70                 1                 0   
537                    70                 9                 7   
538                    80                 1                 0   
539                    70                 7                 6   
540                    70                 9                10   

     Avg. F size (L) (mm) Avg. F size (R) (mm)  Endometrium (mm)  Unnamed: 44  
0                    18.0                 18.0               8.5          NaN  
1                    15.0                 14.0               3.7          NaN  
2                    18.0                 20.0              10.0          NaN  
3                    15.0                 14.0               7.5          NaN  
4                    16.0                 14.0               7.0          NaN  
..                    ...                  ...               ...          ...  
536                  17.5                 10.0               6.7          NaN  
537          

In [7]:
#Show the first 5 rows of the dataset

pcos.head()

Sl. No  Patient File No.  PCOS (Y/N)   Age (yrs)  Weight (Kg)  Height(Cm)   \
0       1                 1           0          28         44.6        152.0   
1       2                 2           0          36         65.0        161.5   
2       3                 3           1          33         68.8        165.0   
3       4                 4           0          37         65.0        148.0   
4       5                 5           0          25         52.0        161.0   

    BMI  Blood Group  Pulse rate(bpm)   RR (breaths/min)  ...  \
0  19.3           15                78                22  ...   
1  24.9           15                74                20  ...   
2  25.3           11                72                18  ...   
3  29.7           13                72                20  ...   
4  20.1           11                72                18  ...   

   Fast food (Y/N)  Reg.Exercise(Y/N)  BP _Systolic (mmHg)  \
0              1.0                  0                  110   
1              0.0                  0                  120   
2              1.0                  0                  120   
3              0.0                  0                  120   
4              0.0                  0                  120   

   BP _Diastolic (mmHg)  Follicle No. (L)  Follicle No. (R)  \
0                    80                 3                 3   
1                    70                 3                 5   
2                    80                13                15   
3                    70                 2                 2   
4                    80                 3                 4   

   Avg. F size (L) (mm) Avg. F size (R) (mm)  Endometrium (mm)  Unnamed: 44  
0                  18.0                 18.0               8.5          NaN  
1                  15.0                 14.0               3.7          NaN  
2                  18.0                 20.0              10.0          NaN  
3                  15.0                 14.0               7.5          NaN  
4                  16.0                 14.0               7.0          NaN  

[5 rows x 45 columns]

In [8]:
#Show the last 5 rows of the dataset

pcos.tail()

Sl. No  Patient File No.  PCOS (Y/N)   Age (yrs)  Weight (Kg)  \
536     537               537           0          35         50.0   
537     538               538           0          30         63.2   
538     539               539           0          36         54.0   
539     540               540           0          27         50.0   
540     541               541           1          23         82.0   

     Height(Cm)    BMI  Blood Group  Pulse rate(bpm)   RR (breaths/min)  ...  \
536      164.592  18.5           17                72                16  ...   
537      158.000  25.3           15                72                18  ...   
538      152.000  23.4           13                74                20  ...   
539      150.000  22.2           15                74                20  ...   
540      165.000  30.1           13                80                20  ...   

     Fast food (Y/N)  Reg.Exercise(Y/N)  BP _Systolic (mmHg)  \
536              0.0                  0                  110   
537              0.0                  0                  110   
538              0.0                  0                  110   
539              0.0                  0                  110   
540              1.0                  0                  120   

     BP _Diastolic (mmHg)  Follicle No. (L)  Follicle No. (R)  \
536                    70                 1                 0   
537                    70                 9                 7   
538                    80                 1                 0   
539                    70                 7                 6   
540                    70                 9                10   

     Avg. F size (L) (mm) Avg. F size (R) (mm)  Endometrium (mm)  Unnamed: 44  
536                  17.5                 10.0               6.7          NaN  
537                  19.0                 18.0               8.2          NaN  
538                  18.0                  9.0               7.3          NaN  
539                  18.0                 16.0              11.5          NaN  
540                  19.0                 18.0               6.9          NaN  

[5 rows x 45 columns]

In [6]:
# List of column names

for col in list(pcos.columns):
    print(col)

Sl. No
Patient File No.
PCOS (Y/N)
 Age (yrs)
Weight (Kg)
Height(Cm) 
BMI
Blood Group
Pulse rate(bpm) 
RR (breaths/min)
Hb(g/dl)
Cycle(R/I)
Cycle length(days)
Marraige Status (Yrs)
Pregnant(Y/N)
No. of abortions
  I   beta-HCG(mIU/mL)
II    beta-HCG(mIU/mL)
FSH(mIU/mL)
LH(mIU/mL)
FSH/LH
Hip(inch)
Waist(inch)
Waist:Hip Ratio
TSH (mIU/L)
AMH(ng/mL)
PRL(ng/mL)
Vit D3 (ng/mL)
PRG(ng/mL)
RBS(mg/dl)
Weight gain(Y/N)
hair growth(Y/N)
Skin darkening (Y/N)
Hair loss(Y/N)
Pimples(Y/N)
Fast food (Y/N)
Reg.Exercise(Y/N)
BP _Systolic (mmHg)
BP _Diastolic (mmHg)
Follicle No. (L)
Follicle No. (R)
Avg. F size (L) (mm)
Avg. F size (R) (mm)
Endometrium (mm)
Unnamed: 44


In [26]:
#gives the size/shape of the dataset
#How many columns and rows are there in the dataset.

num_rows, num_columns = pcos.shape

print("Number of rows:", num_rows)
print("Number of columns:", num_columns)

Number of rows: 541
Number of columns: 45


By analyzing the dataset, we found that there is a column named "unnamed" which contains only null values. This is due to the extra comma after the last column of each row, which is unnecessary and should not be present in our dataset.

In [14]:
#Check the data type of dataframe pcos

pcosdf.dtypes

Sl. No                      int64
Patient File No.            int64
PCOS (Y/N)                  int64
 Age (yrs)                  int64
Weight (Kg)               float64
Height(Cm)                float64
BMI                       float64
Blood Group                 int64
Pulse rate(bpm)             int64
RR (breaths/min)            int64
Hb(g/dl)                  float64
Cycle(R/I)                  int64
Cycle length(days)          int64
Marraige Status (Yrs)     float64
Pregnant(Y/N)               int64
No. of abortions            int64
  I   beta-HCG(mIU/mL)    float64
II    beta-HCG(mIU/mL)     object
FSH(mIU/mL)               float64
LH(mIU/mL)                float64
FSH/LH                    float64
Hip(inch)                   int64
Waist(inch)                 int64
Waist:Hip Ratio           float64
TSH (mIU/L)               float64
AMH(ng/mL)                 object
PRL(ng/mL)                float64
Vit D3 (ng/mL)            float64
PRG(ng/mL)                float64
RBS(mg/dl)    

In [27]:
#Information about the dataset
#This method prints information about a DataFrame including the index dtype and columns, non-null values and memory usage. 

pcosdf.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 541 entries, 0 to 540
Data columns (total 45 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Sl. No                  541 non-null    int64  
 1   Patient File No.        541 non-null    int64  
 2   PCOS (Y/N)              541 non-null    int64  
 3    Age (yrs)              541 non-null    int64  
 4   Weight (Kg)             541 non-null    float64
 5   Height(Cm)              541 non-null    float64
 6   BMI                     541 non-null    float64
 7   Blood Group             541 non-null    int64  
 8   Pulse rate(bpm)         541 non-null    int64  
 9   RR (breaths/min)        541 non-null    int64  
 10  Hb(g/dl)                541 non-null    float64
 11  Cycle(R/I)              541 non-null    int64  
 12  Cycle length(days)      541 non-null    int64  
 13  Marraige Status (Yrs)   540 non-null    float64
 14  Pregnant(Y/N)           541 non-null    in

# Discription

This dataset comprises information on 541 patients, primarily focusing on factors related to polycystic ovary syndrome (PCOS).

The patients have an average age of 31.43 years, with a typical weight of 59.64 kg and height of 156.48 cm. The average Body Mass Index (BMI) among the patients is 24.31, indicating a moderate level of body weight relative to height.

Approximately 32.7% of patients have been diagnosed with PCOS, a hormonal disorder common among women of reproductive age. Additionally, 38.0% of patients are currently pregnant, while the average number of abortions among patients is 0.29.

In [29]:
#If we would like to get a statistical summary of each column, such as count, column mean value, column standard deviation, etc.
#We use the describe method:

pcosdf.describe().T

count        mean          std      min     25%  \
Sl. No                  541.0  271.000000   156.317519    1.000  136.00   
Patient File No.        541.0  271.000000   156.317519    1.000  136.00   
PCOS (Y/N)              541.0    0.327172     0.469615    0.000    0.00   
 Age (yrs)              541.0   31.430684     5.411006   20.000   28.00   
Weight (Kg)             541.0   59.637153    11.028287   31.000   52.00   
Height(Cm)              541.0  156.484835     6.033545  137.000  152.00   
BMI                     541.0   24.307579     4.055129   12.400   21.60   
Blood Group             541.0   13.802218     1.840812   11.000   13.00   
Pulse rate(bpm)         541.0   73.247689     4.430285   13.000   72.00   
RR (breaths/min)        541.0   19.243993     1.688629   16.000   18.00   
Hb(g/dl)                541.0   11.160037     0.866904    8.500   10.50   
Cycle(R/I)              541.0    2.560074     0.901950    2.000    2.00   
Cycle length(days)      541.0    4.940850     1.492020    0.000    4.00   
Marraige Status (Yrs)   540.0    7.680926     4.803835    0.000    4.00   
Pregnant(Y/N)           541.0    0.380776     0.486027    0.000    0.00   
No. of abortions        541.0    0.288355     0.692575    0.000    0.00   
  I   beta-HCG(mIU/mL)  541.0  664.549235  3348.920576    1.300    1.99   
FSH(mIU/mL)             541.0   14.601832   217.022081    0.210    3.30   
LH(mIU/mL)              541.0    6.469919    86.673255    0.020    1.02   
FSH/LH                  541.0    6.904917    60.691982    0.000    1.42   
Hip(inch)               541.0   37.992606     3.967920   26.000   36.00   
Waist(inch)             541.0   33.841035     3.596894   24.000   32.00   
Waist:Hip Ratio         541.0    0.891627     0.046135    0.760    0.86   
TSH (mIU/L)             541.0    2.981281     3.755532    0.040    1.48   
PRL(ng/mL)              541.0   24.321497    14.970391    0.400   14.52   
Vit D3 (ng/mL)          541.0   49.915874   346.206599    0.000   20.80   
PRG(ng/mL)              541.0    0.610945     3.808853    0.047    0.25   
RBS(mg/dl)              541.0   99.835860    18.559298   60.000   92.00   
Weight gain(Y/N)        541.0    0.377079     0.485104    0.000    0.00   
hair growth(Y/N)        541.0    0.273567     0.446202    0.000    0.00   
Skin darkening (Y/N)    541.0    0.306839     0.461609    0.000    0.00   
Hair loss(Y/N)          541.0    0.452865     0.498234    0.000    0.00   
Pimples(Y/N)            541.0    0.489834     0.500359    0.000    0.00   
Fast food (Y/N)         540.0    0.514815     0.500244    0.000    0.00   
Reg.Exercise(Y/N)       541.0    0.247689     0.432070    0.000    0.00   
BP _Systolic (mmHg)     541.0  114.661738     7.384556   12.000  110.00   
BP _Diastolic (mmHg)    541.0   76.927911     5.574112    8.000   70.00   
Follicle No. (L)        541.0    6.129390     4.229294    0.000    3.00   
Follicle No. (R)        541.0    6.641405     4.436889    0.000    3.00   
Avg. F size (L) (mm)    541.0   15.018115     3.566839    0.000   13.00   
Avg. F size (R) (mm)    541.0   15.451701     3.318848    0.000   13.00   
Endometrium (mm)        541.0    8.475915     2.165381    0.000    7.00   

                           50%     75%       max  
Sl. No                  271.00  406.00    541.00  
Patient File No.        271.00  406.00    541.00  
PCOS (Y/N)                0.00    1.00      1.00  
 Age (yrs)               31.00   35.00     48.00  
Weight (Kg)              59.00   65.00    108.00  
Height(Cm)              156.00  160.00    180.00  
BMI                      24.20   26.60     38.90  
Blood Group              14.00   15.00     18.00  
Pulse rate(bpm)          72.00   74.00     82.00  
RR (breaths/min)         18.00   20.00     28.00  
Hb(g/dl)                 11.00   11.70     14.80  
Cycle(R/I)                2.00    4.00      5.00  
Cycle length(days)        5.00    5.00     12.00  
Marraige Status (Yrs)     7.00   10.00     30.00  
Pregnant(Y/N)             0.00   

This dataset provides a comprehensive overview of patient demographics, medical history, laboratory results, and physical characteristics, offering valuable insights into factors relevant to PCOS and gynecological health.